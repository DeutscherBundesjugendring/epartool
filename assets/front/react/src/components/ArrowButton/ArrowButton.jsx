import React from 'react';
import PropTypes from 'prop-types';

const ArrowButton = (props) => {
  const {
    direction,
    disabled,
    label,
  } = props;

  let classes = 'followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count';

  if (direction === 'inward') {
    classes += ' followup-timeline-count-arrow followup-timeline-count-arrow-inward';
  } else {
    classes += ' followup-timeline-count-arrow followup-timeline-count-arrow-outward';
  }

  return (
    <button
      onClick={(e) => {
        e.stopPropagation();
        props.onClick();
      }}
      disabled={disabled}
      className={classes}
      type="button"
    >
      {label}
    </button>
  );
};

ArrowButton.defaultProps = {
  disabled: false,
};

ArrowButton.propTypes = {
  direction: PropTypes.oneOf(['inward', 'outward']).isRequired,
  disabled: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ArrowButton;
