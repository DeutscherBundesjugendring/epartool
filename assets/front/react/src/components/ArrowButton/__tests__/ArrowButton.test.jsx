import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import ArrowButton from '../ArrowButton';


injectTapEventPlugin();

describe('rendering', () => {
  it('renders correctly right', () => {
    const tree = shallow(
      <ArrowButton label="1" direction="outward" onTouchTap={() => {}} />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly left', () => {
    const tree = shallow(
      <ArrowButton label="1" direction="inward" onTouchTap={() => {}} />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly right disabled', () => {
    const tree = shallow(
      <ArrowButton label="1" direction="outward" onTouchTap={() => {}} disabled />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly left disabled', () => {
    const tree = shallow(
      <ArrowButton label="1" direction="inward" onTouchTap={() => {}} disabled />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls onTouchTap', () => {
    const spy = sinon.spy();
    const component = shallow(
      <ArrowButton label="1" direction="outward" onTouchTap={spy} />
    );

    component.simulate('touchTap', { stopPropagation: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });
});
