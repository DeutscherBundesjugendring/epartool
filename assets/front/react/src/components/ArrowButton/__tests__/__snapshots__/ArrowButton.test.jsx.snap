// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly left 1`] = `
<button
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count followup-timeline-count-arrow followup-timeline-count-arrow-inward"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  1
</button>
`;

exports[`rendering renders correctly left disabled 1`] = `
<button
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count followup-timeline-count-arrow followup-timeline-count-arrow-inward"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  1
</button>
`;

exports[`rendering renders correctly right 1`] = `
<button
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count followup-timeline-count-arrow followup-timeline-count-arrow-outward"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  1
</button>
`;

exports[`rendering renders correctly right disabled 1`] = `
<button
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count followup-timeline-count-arrow followup-timeline-count-arrow-outward"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
